[modbus.lambda] Fix inconsistencies in three channel id's

chilobo opened the pull request:
Based on the Stiebel Eltron bundle by Paul Frank I contribute a bundle to manage Lambda Heat Pumps as a OSG addon to the Modbus Binding.

See the README.md for restrictions on usage.

The binding is based on the Modbus description of the manufacturer:
(https://lambda-wp.at/wp-content/uploads/2025/04/Modbus-Beschreibung-und-Protokoll.pdf)
A few parts of the description are not implemented. If you feel you need them, just open a request.

This addition corrects inconsistencies in the file heat-pump-thing-types.xml, changing heatpump prefix to heat-pump. This change affects some channel names of the heat-pump-thing.

chilobo committed:
Updated and signed off version of Lambda Heat Pump

Signed-off-by: Christian Koch <78686276+chilobo@users.noreply.github.com>

chilobo committed:
Updated and signed off version of Lambda Heat Pump

Signed-off-by: Christian Koch <78686276+chilobo@users.noreply.github.com>

chilobo committed:
Merge branch 'openhab:main' into main

lsiepel reviewed (changes requested):
Thanks for the fix.
This would require upgrade instructions for the three changed id's. You can look at the documentation how to do this here: https://www.openhab.org/docs/developer/bindings/thing-xml.html#updating-thing-types
You can also check other bindings to see it in action. For instance `plugwiseha`.

Otherwise LGTM

chilobo commented:
OK, makes sense. 
So I generate a yml-file in the OH-INF/update directory which deletes the old channel names and types and adds the new ones.
Should I add the channels 
`heat-pump-vdaq-swap` and `heat-pump-vdae-swap`
in this file too?

Can I use this pull request or do I have to start a new one?

lsiepel commented:
> OK, makes sense. So I generate a yml-file in the OH-INF/update directory which deletes the old channel names and types and adds the new ones. Should I add the channels `heat-pump-vdaq-swap` and `heat-pump-vdae-swap` in this file too?
> 
> Can I use this pull request or do I have to start a new one?

Not yml, but xml. For the changed channels (only channel type) you can use ‘update-channel’ instruction. For changed is you need both remove and add instructions.
Please also add the swap channels (we should have done that in the other PR).
And yes it should be added in this PR.

chilobo committed:
Updated and signed off version of Lambda Heat Pump

Signed-off-by: Christian Koch <78686276+chilobo@users.noreply.github.com>

chilobo commented:
I added a OH-INF/update/heatpump_instructions.xml;
added the swap-channels;
deleted the misspelled channels heatpump-*;
added the corresponding channels heat-pump-*;
(I guess this is necessary because I didn't just change the channel-type).
I added a thing-type-version part in heat-pump-thing-types.xml.

`plugwiseha` didn't have update-instruction, I found some in `viessmann`, `e3dc`, `fronius`

I am not shure how I can test these changes.

lsiepel commented on bundles/org.openhab.binding.modbus.lambda/src/main/resources/OH-INF/update/heatpump_instructions.xml:
```diff
@@ -0,0 +1,34 @@
+	<thing-type uid="modbus:heat-pump">
+		<instruction-set targetVersion="1">
+			<add-channel id="heat-pump-vdae-swap">
+				<type>typeId="heat-pump:heat-pump-vdae-swap"</type>
```
```suggestion
				<type>modbus:heat-pump-vdae-swap</type>
```

lsiepel commented on bundles/org.openhab.binding.modbus.lambda/src/main/resources/OH-INF/update/heatpump_instructions.xml:
```diff
@@ -0,0 +1,34 @@
+			</add-channel>
+			<remove-channel id="heatpump-request-type"/>
+			<add-channel id="heat-pump-request-type">
+				<type>typeId="heat-pump:heat-pump-request-type"</type>
```
```suggestion
				<type>modbus:heat-pump-request-type</type>
```

lsiepel reviewed (changes requested):
Thansk for adding the Thing upgrade instructions. Testing for this binding is a bit more difficult. But  possible.

Use the shipped binding version, set up the thing and channels. 
Make sure the channel id's are as it was before this PR.
Update the modbus.lambda binding by doing this in the karaf console:

```
bundle:list | grep modbus.lambda 
bundle:update 123 file:/path/to/binding.jar
```

Where 123 is the `bundle id` you got from the first command.

After the bundle is updated, you should see the Thing with updated channel id's without recreating it.

(JFTR, becasue the thingtypeversion is set, this can only be triggered once, or you have to do this on a new openHAB instance)

chilobo committed:
Update bundles/org.openhab.binding.modbus.lambda/src/main/resources/OH-INF/update/heatpump_instructions.xml

Co-authored-by: lsiepel <leosiepel@gmail.com>
Signed-off-by: Christian Koch <78686276+chilobo@users.noreply.github.com>

chilobo committed:
Update bundles/org.openhab.binding.modbus.lambda/src/main/resources/OH-INF/update/heatpump_instructions.xml

Co-authored-by: lsiepel <leosiepel@gmail.com>
Signed-off-by: Christian Koch <78686276+chilobo@users.noreply.github.com>

chilobo committed:
Update bundles/org.openhab.binding.modbus.lambda/src/main/resources/OH-INF/update/heatpump_instructions.xml

Co-authored-by: lsiepel <leosiepel@gmail.com>
Signed-off-by: Christian Koch <78686276+chilobo@users.noreply.github.com>

chilobo committed:
Update bundles/org.openhab.binding.modbus.lambda/src/main/resources/OH-INF/update/heatpump_instructions.xml

Co-authored-by: lsiepel <leosiepel@gmail.com>
Signed-off-by: Christian Koch <78686276+chilobo@users.noreply.github.com>

chilobo committed:
Update bundles/org.openhab.binding.modbus.lambda/src/main/resources/OH-INF/update/heatpump_instructions.xml

Co-authored-by: lsiepel <leosiepel@gmail.com>
Signed-off-by: Christian Koch <78686276+chilobo@users.noreply.github.com>

chilobo commented:
I pulled your changes and tested them on a second system on a Pi4.
It seems the update-instructions were successful.
<img width="1388" height="81" alt="grafik" src="https://github.com/user-attachments/assets/d8c23c7a-44b6-4760-9edb-4fe2c306f576" />

chilobo commented:
Regarding the addon-problem for the modbus-extension:
Should I delete the addons.xml?
Or did you delete the addons.xml in the basic modbus code to fix that problem?

lsiepel commented:
> Regarding the addon-problem for the modbus-extension: Should I delete the addons.xml? Or did you delete the addons.xml in the basic modbus code to fix that problem?

The addon.xml is already removed by another PR, thanks.

chilobo commented:
I corrected errors in units of volumetric flows caused by errors in the Modbus document.
You find the new pull request in:
[https://github.com/openhab/openhab-addons/pull/20596](url)
